src/gt4py/cartesian/gtc/dace/expansion/tasklet_codegen.py

```diff
@@ -62,17 +56,34 @@ def _visit_offset(
         res = dace.subsets.Range([r for i, r in enumerate(ranges.ranges) if int_sizes[i] != 1])
         return str(res)
 
-    def visit_CartesianOffset(self, node: common.CartesianOffset, **kwargs: Any) -> str:
+    def _explicit_indexing(
+        self, node: common.CartesianOffset | dcir.VariableKOffset, **kwargs: Any
+    ) -> str:
+        """If called from the explicit pass we need to be add manually the relative indexing"""
+        return f"__k+{self.visit(node.k, **kwargs)}"
+
+    def visit_CartesianOffset(
+        self, node: common.CartesianOffset, explicit=False, **kwargs: Any
+    ) -> str:
+        if explicit:
+            return self._explicit_indexing(node, **kwargs)
+
         return self._visit_offset(node, **kwargs)
 
-    def visit_VariableKOffset(self, node: dcir.VariableKOffset, **kwargs: Any) -> str:
+    def visit_VariableKOffset(
+        self, node: dcir.VariableKOffset, explicit=False, **kwargs: Any
+    ) -> str:
+        if explicit:
```

Comment:
@FlorianDeconinck are we planning on supporting explicit, variable offsets?

Reply:
Hmm, this might be aggregating more than Variable "offset", but variable access altogether. 

So we shouldn't, but I wonder if the implementation is not aggregating too much use case here